test/integ/image_integ_test.js

```diff
@@ -0,0 +1,205 @@
+                            });
+                            return;
+                        }
+                        resolve(res.body);
+                        agent.close();
+                    })
+                    .catch((err) => {
+                        reject(err);
+                        agent.close();
+                    });
+            });
+        }
+
+        before(function () {
+            return new Promise(async (resolve) => {
+                mongod = new MongoMemoryServer();
+                await mongod.getUri();
+                await mongod.getPort();
+                await mongod.getDbPath();
+                await mongod.getDbName();
+                const testDBURL = mongod.getInstanceInfo().uri;
+                db = await MongoClient.connect(testDBURL);
+                databaseOps.startDatabaseClient(function (err) {
+                    if (err) {
+                        console.error(err);
+                        return;
+                    }
+    
+                    databaseOps.addUser({
+                        username: "test-user",
+                        password: "test",
+                        email: "test@test.com"
+                    }, () => {
+                        integTestUtils.addImagesForUser([
+                            {
+                                fileName: "PNGtest.png",
+                                mimeType: "image/png",
+                            },
+                            {
+                                fileName: "JPEGtest.jpg",
+                                mimeType: "image/jpeg",
+                            },
+                            {
+                                fileName: "BMPtest.bmp",
+                                mimeType: "image/bmp",
+                            },
+                            {
+                                fileName: "GIFtest.gif",
+                                mimeType: "image/gif",
+                            },
+                        ], 'test-user')
+                            .then((imgs) => {
+                                imgs.forEach(imgResult => {
+                                    imagesLookupByExt.set(getImageExt(imgResult.mimetype), imgResult);
+                                });
+                                resolve();
+                            });
+                    });
+                }, {
+                    dbURL: testDBURL
+                });
+            });
+        });
+
+        beforeEach((done) => {
+            databaseOps.addUser({
+                username: TEST_USER,
+                password: "test",
+                email: "test@test.com"
+            }, () => {
+                agent = getServerAgent();
+                done();
+            });
+        });
+
+        afterEach(() => {
+            usersCollection = db.collection("users");
+            usersCollection.deleteMany({});
+            agent.close();
+        });
+
+        after(() => {
```

Comment:
I may want to look into moving all these setup methods into its own util structure in the future, so less code is duplicated across tests.

Reply:
Will address this in another PR